### DISCUSSION TITLE
Sensor Error - automatic disable collection - Dexcom G6

### DISCUSSION THREAD

**Discussion by lmiekus** [NightscoutFoundation/xDrip#2213] [Ideas]
I would like to add new feature in xdrip to automaticly set xdrip from hardware data source -> G5/G6 to hardware data source -> disable collection in the situation when transimtter starts reporting sensor error. I've noticed that G6 much quicker restores to normal condition upon reporting sensor error when xdrip stops collecting BG data by setting it to disable collection. 

I am looking for a way to set xdrip automaticly into disable collection upon receving senssor error from transmiter and after 30 min from that time switching back to G5/G6. At the moment this whole process needs to be done manualy. Thx in advanced for your help.

**Comment by Navid200** [comment]
The sensor error message, as far as I know, comes from the transmitter.
Are you suggesting that the transmitters itself will recover faster if there is no receiving device?
The problem with that assertion is that it will be very difficult to prove.

The other problem is that if the sensor error is due to blood having leaked too much into the contacts, it is possible to remove the transmitter and clean the contacts and effectively recover from the situation.  But, if it lasts too long, it may not be possible to recover any longer.  If we automatically stop collecting as soon as we receive a sensor error, we may lose the option to clean and recover.

There are also times a sensor is too old and the resulting noise is what the transmitter does not like.  In that case, what would be the point of disabling collection?

Can you think of a way to convince yourself this is not just a coincidence that disabling collection can address sensor issues?
